### PR TITLE
Use sbt-bom

### PR DESCRIPTION
**Pull request by RustedBones**
Leverage `sbt-bom` [plugin](https://github.com/heremaps/here-sbt-bom) and override versions for
- `beam`
- `gcp`
- `jackson`
- `netty`

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -445,66 +432,78 @@ lazy val unusedCompileDependenciesTestSkipped = Def.task {
...
+    resolvers ++= Resolver.sonatypeOssRepos("public"),
+    excludeDependencies += Exclude.beamKafka,
+    excludeDependencies ++= Exclude.loggerImplementations,
+    dependencyOverrides ++= beamBom.key.value.bomDependencies ++
+      gcpBom.key.value.bomDependencies ++
+      jacksonBom.key.value.bomDependencies ++
+      nettyBom.key.value.bomDependencies ++
+      Seq(
+        // slf4j-bom only available for v2
+        "org.slf4j" % "slf4j-api" % slf4jVersion
```
Comment (RustedBones) [inline build.sbt:480]: we absolutely need to downgrade to slf4j v1 as some deps are pulling v2

File: build.sbt [hunk 2]
```diff
@@ -694,8 +687,8 @@ lazy val `scio-core` = project
       "com.github.ben-manes.caffeine" % "caffeine" % caffeineVersion % Provided,
       "com.google.apis" % "google-api-services-dataflow" % googleApiServicesDataflowVersion % Provided,
       "org.apache.beam" % s"beam-runners-flink-$flinkMinorVersion" % beamVersion % Provided,
-      "org.apache.beam" % "beam-runners-google-cloud-dataflow-java" % beamVersion % Provided,
-      "org.apache.beam" % s"beam-runners-spark-$sparkMajorVersion" % beamVersion % Provided,
+      "org.apache.beam" % "beam-runners-google-cloud-dataflow-java" % beamVersion % Provided excludeAll (Exclude.avro),
```
Comment (shnapz) [inline build.sbt:690]: why do you excluding avro in this PR?

Comment (RustedBones) [reply build.sbt:702]: I'm not using `dependencyOverrides` for avro anymore to check the pulled version is the expected one. This means we need to exclude also in non compile scope.

I'm actually wondering if we should go with the flow an let OSS scio upgrade to latest avro WDYT ?

Comment (RustedBones) [reply build.sbt]: Since #5415, no need to evict

**Pull request merged**